ingest/monitoring/mixpanel_log.py

```diff
@@ -0,0 +1,74 @@
+from timeit import default_timer as timer
+from typing import List, Dict  # noqa: F401
+import functools
+from contextlib import ContextDecorator
+
+from config import add_child_event
+from monitor import testing_guard
+
+
+class MetricTimedNode(ContextDecorator):
+    """Context manager that logs properties and performance times of Mixpanel events."""
+
+    def __init__(
+        self,
+        function_name,
+        perf_time_name,
+        metric_properties: Dict,
+        props: Dict = None,
+        child_event_name: str = None,
+    ):
+        props["functionName"] = function_name
+        self.child_event_name = child_event_name
+        self.props = props
+        self.perf_time_name = perf_time_name if perf_time_name else "perfTime"
```

Comment:
This ia little hard to explain. But basically the default perfTime name is perfTime. For child events, the perfTime would be named after the child event. For example, the [perfTime for sorted MTX files](https://github.com/broadinstitute/scp-ingest-pipeline/blob/39e1715c44c68f4f81c553ec593615b1ce6864dc/ingest/expression_files/mtx.py#L209) is named  "isSortedPerfTime". I could add a comment or maybe do a slight refactor such as:

```
default_perfTime_name = "perfTime"
perf_time_name = perf_time_name if perf_time_name else default_perfTime_name
```

Reply:
Your original logic seems clear, I just think it'd just be clearer and less code to use default parameter values in the method signature.

Also (more importantly), please use the form `perfTime:<foo>` instead of `<foo>PerfTime`  (e.g. `perfTime:isSorted` instead of `isSortedPerfTime`) for consistency with such logging in `single_cell_portal_core`.

Reply:
Just for my understanding perfTime is a number. So the perfTimes of child events are named like <child event name>PerfTime. A single ingest run captures all perfTimes of decorated functions. Please see the example [here](https://github.com/broadinstitute/scp-ingest-pipeline/pull/161#issue-511678735). The field value "perfTime" captures the total amount it took to parse and store.

Reply:
0f77b31

Reply:
Yup, `perfTime` is the entire time as a number -- including the sum of all children.

But for consistency with other logging, those child perfTimes property names should take the form `perfTime:<child event name>` (e.g. `"perfTime:isSorted": 1234`) and not `<child event name>PerfTime`.